usr/src/test/zfs-tests/tests/functional/reservation/cleanup.ksh

```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/ksh -p
+#!/bin/ksh -p
```

Comment:
Why?  Other test cases use ```#! /usr/bin/ksh -p``` and it's what we have in usr/src/prototypes.

Reply:
because the same on Debian - it is available at #!/bin/ksh as others shell and it is available in /etc/shells:
```# /etc/shells: valid login shells
/bin/sh
/bin/dash
/bin/bash
/bin/rbash
/bin/ksh93
/bin/rksh93

Reply:
So if I have my own distro with ksh installed in /opt/ksh, I can freely "fix" all of the test cases in openzfs repository to have ```#!/opt/ksh/bin/ksh``` shebang?

Reply:
Well, at least change *all* of those then, and put it out for separate review.

Reply:
I agree with @yuripv, it would be nice to have them all be the same. I do have a preference for `#!/usr/bin/ksh` because it matches what's in the prototypes file.

Reply:
It may not be relevant, but for what it's worth on Linux we've needed to change these all to `#!/bin/ksh`.  This is a change we always need to make work porting OpenZFS test cases.

Reply:
@behlendorf and your vote to use `#!/bin/ksh` for KSH scripts as default?

Reply:
I'd be fine with that too. The main benefit is that it's the same everyplace in the suite.

Reply:
Selfishly that would be my preference, then we won't need to remember to make this change when porting for ZoL.  But I don't know enough about the OpenZFS repository to say what's best for that platform.

Reply:
@jwk404 based on my grep output - it is not the same for all places - probably it was fixed to be more applicable between platforms to use `#!/bin/ksh`
and to me personally - `#!/bin/ksh` will be working on all platforms.

Reply:
A change such as this should be discussed on the illumos-dev list before considering integration.

@ikozhukhov Have you considered changing the tests so that the shebang is determined at build time?

Reply:
these changes more related to OpenZFS and should be discussed/approved on related topic - because they are using outside illumos a lot.
no - i'm not proposing to update this shebang by build time - it should be static.
if this change no applicable to illumos - we need agreements for OpenZFS project and we can fork zfs tests to separate project/repo

Reply:
We already fix the shebangs for python scripts: https://github.com/illumos/illumos-gate/blob/master/usr/src/Makefile.master#L267

Doing the same for the tests should be easy.

Reply:
шt can be separate issue and you can do it, but it is not related to current update

Reply:
The other way round, actually, this change is totally unrelated to what the synopsis describes.

For the record, I'm NOT against changing the path in shebang, you just did it for small part of the scripts in the zfs-tests, and under unrelated issue id.

Reply:
take a look grep what i posted early - my changes based on others shebang updates, but to be more consistent with others platforms like zfsonlinux - and they are no need update these tests again on their repo.

Reply:
I'm done here as you aren't reading what others are saying; -1 from me (if that matters at all).

Reply:
try to understand : ZFS is not on illumos only now - and have to look and hear other projects - because they are needs try to find solutions with: how to play/adopt updates every time.
you try dictate rules based on illumos platform, but zfs code is shared between others and rules should be changed.

Reply:
Igor, do you plan to pursue this, or should we close this PR? I think if you want to pursue it, the change should make the #! line consistent throughout the suite, and effort should be undertaken to ensure that the change isn't inconvenient to downstream consumers.

Reply:
John, you can close this PR. 
also, i'd like to see the rules for zfs tests, because based on grep report i can see a lot of inconsistency. All tests should be fixed to use the same rules.
also, we are interested in the same zfs tests code base for illumos & ZoL based environments, but as i can see - it is impossible right now, because illumos rules incompatible with ZoL.